### PR TITLE
fix filters and add tests

### PR DESCRIPTION
**Pull request by ming535**

### CODE REVIEW

File: gravity/filters/delete_dml_column_filter.go [hunk 1]
```diff
@@ -49,6 +55,23 @@ func (f *deleteDmlColumnFilter) Filter(msg *core.Msg) (continueNext bool, err er
 		return false, errors.Errorf("DmlMsg is null")
 	}
 
+	if msg.DmlMsg.PkColumns != nil {
+		var newPkColumns []string
+		for _, oldName := range msg.DmlMsg.PkColumns {
+			for _, name := range f.columns {
+				_, ok := msg.DmlMsg.Pks[name]
```
Comment (Ryan-Git) [inline gravity/filters/delete_dml_column_filter.go]: why check `Pks` again? Can't we remove it directly? `Pks` and `PkColumns` are always in sync I think.

File: pkg/utils/type_cast.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
+package utils
+
+import (
+	"reflect"
+
+	"github.com/juju/errors"
+)
+
+func CastToSlice(arg interface{}) (out []interface{}, ok bool) {
```
Comment (Ryan-Git) [inline pkg/utils/type_cast.go:10]: Why not cast directly? out, ok := arg.([]interface{})

**Review by Ryan-Git — CHANGES_REQUESTED** [review]

File: gravity/filters/delete_dml_column_filter.go (hunk 1 above)
Comment (ming535) [reply gravity/filters/delete_dml_column_filter.go]: It's not easy to delete an element in a slice by the element's content.

Comment (Ryan-Git) [reply gravity/filters/delete_dml_column_filter.go]: I mean `_, ok := msg.DmlMsg.Pks[name]` seems unnecessary.

File: pkg/utils/type_cast.go (hunk 2 above)
Comment (ming535) [reply pkg/utils/type_cast.go:10]: just to make sure that `arg` can be any type of slice, for example `[]string`, `[]interface{}`, useful for both test and actual code.

File: gravity/filters/delete_dml_column_filter.go (hunk 1 above)
Comment (ming535) [reply gravity/filters/delete_dml_column_filter.go]: if `f.columns` have column name that is not a primary key, then there will be a problem

Comment (Ryan-Git) [reply gravity/filters/delete_dml_column_filter.go]: don't quite understand. if `f.columns` doesn't contain `oldName`, then `oldName` should be kept, is it right?

Comment (Ryan-Git) [reply gravity/filters/delete_dml_column_filter.go]: what if `columns` is ['pk2', 'pk1', 'r'], `PkColumns` is ['pk1', 'pk2', 'pk3'], then `newPkColumns` would be ['pk1', 'pk2'], which is wrong.

**Review by Ryan-Git — APPROVED** [review]

**Pull request merged**